WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -419,5 +419,33 @@ private async Task<EstimateSmartFeeResponse> GetEstimateSmartFeeAsync(int target
 
 			return feeResponse;
 		}
+
+		[HttpGet("status")]
+		[ProducesResponseType(typeof(StatusResponse), 200)]
+		[ResponseCache(Duration = 10, Location = ResponseCacheLocation.Client)]
+		public async Task<StatusResponse> GetStatusAsync()
+		{
+			var result = new StatusResponse();
+
+			try
+			{
+				var lastFilter = Global.IndexBuilderService.GetLastFilter();
+				var lastFilterHash = lastFilter.Header.BlockHash;
+
+				var bestHash = await RpcClient.GetBestBlockHashAsync();
+				var lastBlock = await RpcClient.GetBlockAsync(bestHash);
```

Comment:
This is a bit crazy @molnard because it opens the doors to DoS attacks. I think it could be mitigated by caching the result for 5 to 10 seconds.

Reply:
Right, added caching!